build/yamls/base/crds.yml

```diff
@@ -667,9 +919,51 @@ spec:
                           type: object
                           properties:
                             podSelector:
-                              x-kubernetes-preserve-unknown-fields: true
+                              properties:
+                                matchExpressions:
+                                  type: array
+                                  items:
+                                    type: object
+                                    properties:
+                                      key:
+                                        type: string
+                                      operator:
+                                        enum:
+                                          - In
+                                          - NotIn
+                                          - Exists
+                                          - DoesNotExist
+                                        type: string
+                                      values:
+                                        type: array
+                                        items:
+                                          type: string
+                                matchLabels:
+                                  type: object
+                              type: object
                             namespaceSelector:
-                              x-kubernetes-preserve-unknown-fields: true
+                              properties:
+                                matchExpressions:
+                                  type: array
+                                  items:
+                                    type: object
+                                    properties:
+                                      key:
+                                        type: string
+                                      operator:
+                                        enum:
+                                          - In
+                                          - NotIn
+                                          - Exists
+                                          - DoesNotExist
+                                        type: string
+                                      values:
+                                        type: array
+                                        items:
+                                          type: string
+                                matchLabels:
+                                  type: object
+                              type: object
                             externalEntitySelector:
                               x-kubernetes-preserve-unknown-fields: true
```

Comment:
Could you do the same for `externalEntitySelector`?

Reply:
externalEntitySelector validations added